docs/scenario_plugin_api.md

```diff
@@ -0,0 +1,131 @@
+# Scenario Plugin API:
+
+This API enables seamless integration of Scenario Plugins for Krkn. Plugins are automatically 
+detected and loaded by the plugin loader, provided they extend the `AbstractPluginScenario` 
+abstract class, implement the required methods, and adhere to the specified [naming conventions](#naming-conventions).
+
+## Plugin folder:
+
+The plugin loader automatically loads plugins found in the `krkn/scenario_plugins` directory, 
+relative to the Krkn root folder. Each plugin must reside in its own directory and can consist 
+of one or more Python files. The entry point for each plugin is a Python class that extends the 
+[AbstractPluginScenario](../krkn/scenario_plugins/abstract_scenario_plugin.py) abstract class and implements its required methods.
+
+## `AbstractPluginScenario` abstract class:
+
+This [abstract class](../krkn/scenario_plugins/abstract_scenario_plugin.py) defines the contract between the plugin and krkn.
+It consists of two methods:
+- `run(...)`
+- `get_scenario_type()`
+
+Most IDEs can automatically suggest and implement the abstract methods defined in `AbstractPluginScenario`:
+![pycharm](scenario_plugin_pycharm.gif)
+_(IntelliJ PyCharm)_
+
+### `run(...)`
+
+```python
+    def run(
+        self,
+        run_uuid: str,
+        scenarios_list: list[str],
+        krkn_config: dict[str, any],
+        telemetry: KrknTelemetryOpenshift,
+    ) -> tuple[list[str], list[ScenarioTelemetry]]:
+
+```
+
+This method represents the entry point of the plugin and the first method 
+that will be executed.
+#### Parameters:
+
+- `run_uuid`:
+  - the uuid of the chaos run generated by krkn for every single run.
+- `scenarios_list`:
+  - the list of the paths of every scenario config file defined in the `config.yaml` for the `scenario type` that implements the plugin.
+- `krkn_config`:
+  - the full dictionary representation of the `config.yaml`
+- `telemetry`
+  - it is a composite object of all the [krkn-lib](https://krkn-chaos.github.io/krkn-lib-docs/modules.html) objects and methods needed by a krkn plugin to run.
+ 
+### Return value(s):
+The run method is expected to return a tuple of:
+- `list[str]`:
+  - the list of the scenarios that failed in the so called: _[game-loop](#game-loop)_ (pet name :-)
+- `list[ScenarioTelemetry]`:
+  - the list of the [ScenarioTelemetry](https://krkn-chaos.github.io/krkn-lib-docs/models.telemetry.models.html#models.telemetry.models.ScenarioTelemetry) of each scenario executed in the _[game loop](#game-loop)_
+
+### Game Loop:
```

Comment:
Not sure what is meant here by "Game Loop", can you elaborate some so we could specify a different title

Reply:
https://gameprogrammingpatterns.com/game-loop.html

It is a term that comes from game development, it's a bit of stretch but I wanted to give it a name because it is somehow mandatory and I cannot enforce this pattern at code... or maybe yes! Let me think about it better!!